src/actions/PatientActions.js

```diff
@@ -27,27 +27,94 @@ const editPatient = patient => ({
 
-  if (currentPatient) {
-    const { addressOne: line1, addressTwo: line2 } = completedForm;
-    const { firstName = '', lastName = '', billingAddress = {} } = currentPatient;
-    const patientName = `${lastName}, ${firstName}`;
-
-    UIDatabase.write(() => {
-      UIDatabase.update('Address', { ...billingAddress, line1, line2 });
-      UIDatabase.update('Name', { ...currentPatient, ...completedForm, name: patientName });
-    });
-  } else {
-    UIDatabase.write(() => {
-      createRecord(UIDatabase, 'Patient', completedForm);
-    });
-  }
+  const {
+    id: currentPatientId,
+    code: currentCode,
+    name: currentName,
+    firstName: currentFirstName,
+    lastName: currentLastName,
+    dateOfBirth: currentDateOfBirth,
+    emailAddress: currentEmailAddress,
+    phoneNumber: currentPhoneNumber,
+    billingAddress: currentBillingAddress,
+    country: currentCountry,
+    supplyingStoreId: currentSupplyingStoreId,
+    isActive: currentIsActive,
+    female: currentFemale,
+  } = currentPatient ?? {};
+
+  const {
+    id: currentBillAddressId,
+    line1: currentLine1,
+    line2: currentLine2,
+    line3: currentLine3,
+    line4: currentLine4,
+    zipCode: currentZipCode,
+  } = currentBillingAddress ?? {};
+
+  const {
+    id: patientId,
+    code: patientCode,
+    firstName: patientFirstName,
+    lastName: patientLastName,
+    dateOfBirth: patientDateOfBirth,
+    emailAddress: patientEmailAddress,
+    phoneNumber: patientPhoneNumber,
+    addressOne: patientLine1,
+    addressTwo: patientLine2,
+    country: patientCountry,
+    female: patientFemale,
+  } = patientDetails ?? {};
+
+  const id = patientId ?? currentPatientId;
+  const code = patientCode ?? currentCode;
+  const firstName = patientFirstName ?? currentFirstName;
+  const lastName = patientLastName ?? currentLastName;
+  const name = `${lastName}, ${firstName}` || currentName;
+  const dateOfBirth = patientDateOfBirth ?? currentDateOfBirth;
+  const emailAddress = patientEmailAddress ?? currentEmailAddress;
+  const phoneNumber = patientPhoneNumber ?? currentPhoneNumber;
+  const billAddressId = currentBillAddressId;
+  const billAddress1 = patientLine1 ?? currentLine1;
+  const billAddress2 = patientLine2 ?? currentLine2;
+  const billAddress3 = currentLine3;
+  const billAddress4 = currentLine4;
+  const billPostalZipCode = currentZipCode;
+  const country = patientCountry ?? currentCountry;
+  const female = patientFemale ?? currentFemale;
+  const supplyingStoreId = currentSupplyingStoreId;
+  const isActive = currentIsActive;
+
+  const patientRecord = {
+    id,
+    code,
+    firstName,
+    lastName,
+    name,
+    dateOfBirth,
+    emailAddress,
+    phoneNumber,
+    billAddressId,
+    billAddress1,
+    billAddress2,
+    billAddress3,
+    billAddress4,
+    billPostalZipCode,
+    country,
+    female,
+    supplyingStoreId,
+    isActive,
+  };
+
+  UIDatabase.write(() => createRecord(UIDatabase, 'Patient', patientRecord));
```

Comment:
I find code like this very interesting - you want to be explicit, but that's so the code is clear and easy to read, but then if it's too long and explicit, it becomes harder to read!

I haven't checked every field but I think something a little more JS could make it a little easier on the eyes

```
const record = { ...currentPatient, ...{... patientDetails, suppliyingStore: UID..(thisstore), name: ... },  
```

Reply:
Yeah, should have been more descriptive in the PR. For now I'm keeping it explicit to make it clear exactly what params are coming from where and going where. The probably with the nice spread syntax is that you can't easily see what parameters are actually being used (I had to cross-reference other files and it was easier in the end just to log everything).  

To answer the question re. explicitness being harder to read, I think key point is that explicitness isn't to make the code nicer or easier to read (it's definitely harder to read and uglier!), the point is to show the data being passed around. IMO, in this case using spread syntax makes it nicer to look at, but doesn't make it easier to read, as the information is all hidden, so it has to be found by reading other methods and logging etc. 

Saying all that, the plan is still to make it nice and tidy once all the reviewing and stuff is done (i.e. at the point where functionality is working well and there's less need for that data to be explicit).